subprojects/docs/src/docs/userguide/workingWithDependencies.adoc

```diff
@@ -15,73 +15,47 @@
 [[working_with_dependencies]]
 == Working with Dependencies
 
-For the examples below we have the following dependencies setup:
+Gradle provides an extensive API for navigating, inspecting and post-processing dependencies and artifacts of resolved modules. With this API at hand, you can implement more complex requirements to your build requiring a "runtime view" of dependencies.
 
-++++
-<sample id="configurationHandlingSetup" dir="userguide/artifacts/configurationHandling" title="Configuration.copy">
-            <sourcefile file="build.gradle" snippet="setup"/>
-        </sample>
-++++
-
-The dependencies have the following transitive dependencies:
-
-shark-1.0 -&gt; seal-2.0, tuna-1.0
+=== Iterating over dependencies assigned to a configuration
 
-orca-1.0 -&gt; seal-1.0
+The <<sec:listing_dependencies,dependencies report>> is a built-in task that comes with every project independent of the applied plugins. Sometimes you'll want to implement logic based on the resolved dependencies of a project e.g. for building a plugin that visualizes the dependency graph as an image. The api:org.gradle.api.artifacts.Configuration[] API provides a wealth of methods to navigate and inspect dependencies assigned to a configuration.
 
-tuna-1.0 -&gt; herring-1.0
-
-You can use the configuration to access the declared dependencies or a subset of those:
+You can iterate over complete set of dependencies assigned to a configuration with the help of the method api:org.gradle.api.artifacts.Configuration#getDependencies()[].
 
 ++++
-<sample id="configurationHandlingDependencies" dir="userguide/artifacts/configurationHandling" title="Accessing declared dependencies">
-            <sourcefile file="build.gradle" snippet="dependencies"/>
-            <output args="-q dependencies"/>
-        </sample>
+<sample id="iterating-dependencies" dir="userguide/dependencies/iteratingDependencies" title="Iterating over the dependencies assigned to a configuration">
+    <sourcefile file="build.gradle" snippet="iteration-task" />
+</sample>
 ++++
 
-The `dependencies` task returns only the dependencies belonging explicitly to the configuration. The `allDependencies` task includes the dependencies from extended configurations.
-
-To get the library files of the configuration dependencies you can do:
+=== Iterating over artifacts resolved for a module
 
-++++
-<sample id="configurationHandlingAllFiles" dir="userguide/artifacts/configurationHandling" title="Configuration.files">
-            <sourcefile file="build.gradle" snippet="allFiles"/>
-            <output args="-q allFiles"/>
-        </sample>
-++++
+None of the <<inspecting_dependencies,dependency reporting>> helps you with inspecting or further processing the underlying, resolved artifacts of a module. A typical use case for wanting to access the artifacts is to copy them into a specific directory or filter out files of interest based on a specific file extension.
```

Comment:
"use case for wanting to access the artifacts" -> "use case for accessing the artifacts"

Reply:
done